nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/QueryCassandra.java

```diff
@@ -277,7 +279,9 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             final ResultSet resultSet;
 
             if (queryTimeout > 0) {
-                resultSet = connectionSession.execute(selectQuery, queryTimeout, TimeUnit.MILLISECONDS);
+                Statement statement = new SimpleStatement(selectQuery);
+                statement.setReadTimeoutMillis((int) queryTimeout);
```

Comment:
This cast to `(int)` is not necessary.
```suggestion
                statement.setReadTimeoutMillis(queryTimeout);
```